### DISCUSSION TITLE
Is there a way to perform an inverse transformation on data that is not in tio format?

### DISCUSSION THREAD

**Discussion by Bigsealion** [TorchIO-project/torchio#1168] [Q&A] [answered]
I want to using `tio` in a code to implement test-time augmentation. I hope to directly applying `tio` to augment the input data (numpy), obtain segmentation results, and then inverse-transform the segmentation results back to the original space.

I referred to this [website,](https://torchio.readthedocs.io/transforms/transforms.html#:~:text=transformed_native_space%20%3D%20segmentation.apply_inverse_transform(image_interpolation%3D%27linear%27)) but this code snippet doesn't seem to work.

The code I expect is as follows. 
```
import torchio as tio
tio_transform = tio.Compose([
    tio.RandomFlip(axes=(0, 1, 2), p=0.5),
    tio.RandomAffine(scales=(0.85, 1.25), degrees=15, p=0.2, isotropic=False, default_pad_value="mean"),
    tio.RandomGamma(log_gamma=(-0.3, 0.3), p=0.3)
    ])

# here, img is a numpy array (or torch.tensor) that has already been loaded.
img_aug = tio_transform(img)
# output predicted mask by a trained model
mask_predicted = model(img_aug)

# Here is what I hope to achieve, but I'm not sure how to do it.
mask_in_origin_space = mask_predicted.apply_inverse_transform(image_interpolation='linear')
```

Is there a way to achieve this? Thank you!

**Comment by romainVala** [comment, marked as answer]
Hello
the parameters, needed for the inverse_transform are stored in torchio subject (that has been transformed)
So you need to go with a torchio subject, you can adapt your code like that:

```
array = np.random.rand(1, 256, 256, 159)
suj_tio = torchio.Subject(t1=torchio.ScalarImage(tensor=array))
suj_aug = tio_transform(suj_tio)

mask_predicted = model(suj_aug.t1.data)
suj_aug.add_image(torchio.ScalarImage(tensor = mask_predicted),'predicted_mask')
suj_back = suj_aug.apply_inverse_transform() #so it should apply to both image in subject (t1 and predicted_mask)

```

**Reply by Bigsealion** [reply]
Thank you for your response, it's very helpful!
